### PR TITLE
notion-app-enhanced: init at 2.0.16-5

### PR DESCRIPTION
**Pull request by sei40kr**
###### Motivation for this change

Add notion-app-enhanced, a rebuild of the [Notion](https://www.notion.so) official client with Linux support & some improvements.

There're macOS & arm64 Linux supports, but I couldn't test them to work.
So I created this PR without them.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/notion-app-enhanced/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  meta = with lib; {
+    description = "Notion Desktop builds with Notion Enhancer for Windows, MacOS and Linux.";
+    homepage = "https://github.com/notion-enhancer/desktop";
```
Comment (sei40kr) [inline pkgs/applications/office/notion-app-enhanced/default.nix:27]: The repositories under [notion-enhancer](https://github.com/notion-enhancer) are all MIT licensed, but they contains official Notion client source code, so I set here to `licenses.unlicense`.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -33694,4 +33694,6 @@ with pkgs;
   };
 
   zthrottle = callPackage ../tools/misc/zthrottle { };
+
+  notion-app-enhanced = callPackage ../applications/office/notion-app-enhanced { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33698]: Please sort the entry into a fitting place.

File: pkgs/applications/office/notion-app-enhanced/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/office/notion-app-enhanced/default.nix:27]: Do they not contain the source code? If not we need to set the license to unfree not unlicense (which is a license name).

Comment (sei40kr) [reply pkgs/applications/office/notion-app-enhanced/default.nix:27]: @SuperSandro2000 
> Do they not contain the source code? If not we need to set the license to unfree not unlicense (which is a license name).

Thanks you. I changed it to `unfree`.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (sei40kr) [reply pkgs/top-level/all-packages.nix]: @SuperSandro2000 
Fixed.

**Comment by WolfangAukang** [comment]
#141238

**Pull request merged**